components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Response.java

```diff
@@ -12,25 +12,76 @@
  */
 package org.sonatype.nexus.repository.view;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 import org.sonatype.nexus.common.collect.AttributesMap;
 
 import com.google.common.collect.Maps;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
 
 /**
  * View response.
  *
+ * @see Response.Builder
  * @since 3.0
  */
 public class Response
 {
-  private Status status;
+  private final AttributesMap attributes;
+
+  private final Headers headers;
+
+  private final Status status;
+
+  private final Payload payload;
+
+  private Response(final AttributesMap attributes,
+                   final Headers headers,
+                   final Status status,
+                   @Nullable final Payload payload)
+  {
+    this.attributes = attributes;
+    this.headers = headers;
+    this.status = status;
+    this.payload = payload;
+  }
+
+  public AttributesMap getAttributes() {
+    return attributes;
+  }
+
+  public Headers getHeaders() {
+    return headers;
+  }
+
+  public Status getStatus() {
+    return status;
+  }
+
+  @Nullable
+  public Payload getPayload() {
+    return payload;
+  }
+
+  @Override
+  public String toString() {
+    return getClass().getSimpleName() + "{" +
+        "status=" + status +
+        ", payload=" + payload +
+        '}';
+  }
+
+  //
+  // Attributes
+  //
 
   /**
    * Custom attributes to configure backing and logging.
    */
-  private static class Attributes
+  public static class Attributes
```

Comment:
I don't see a method that returns this type, so is this being made public for testing or some other purpose?